tests/components/vizio/test_config_flow.py

```diff
@@ -291,6 +303,29 @@ def vizio_cant_connect_fixture():
     assert result["data"][CONF_VOLUME_STEP] == VOLUME_STEP
 
 
+async def test_import_flow_update_optons(
```

Comment:
How do I assert that the options were actually updated? I tried to get the entry from the first configuration to check the options value but it stays `None` even though I update the options value during import, and testing it using the frontend seems to work successfully

Reply:
The options are updated in the vizio media_player platform set up. We're not waiting for that to happen.

We can use `hass.async_block_till_done` to wait. Make sure all I/O is patched in the platform set up.

Reply:
There's a typo in this test name.

options

Reply:
Ah I see. I have done that and made it further, but now I get the following error. Do I need to patch this call too or is there a way to bypass the service call in the `config_flow` as I think this should only get called during the discovery workflow:
```
Traceback (most recent call last):
  File ".../home-assistant/homeassistant/core.py", line 1201, in async_call
    raise ServiceNotFound(domain, service) from None
homeassistant.exceptions.ServiceNotFound: Unable to find service persistent_notification/create
ERROR:homeassistant.core:Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File ".../home-assistant/homeassistant/core.py", line 1201, in async_call
    raise ServiceNotFound(domain, service) from None
homeassistant.exceptions.ServiceNotFound: Unable to find service persistent_notification/dismiss
```

Reply:
Also after I import the config a second time in the test with a new `volume_step` value, I am having trouble accessing the ConfigEntry. `hass.config_entries.async_get_entry(UNIQUE_ID)` returns `None` after the update, and I tried saving the entry from the result of creating the entry initially, which also returns `None` after the update. Any ideas?

Reply:
It's nice to patch the persistent notification, yes. It doesn't really matter for the test though. But I recommend doing it to have a clean log. It makes it easier to troubleshoot.

Reply:
`config_entries.async_get_entry` needs the entry id as parameter, not the unique_id of the entry. Those are two different things.

Reply:
https://github.com/home-assistant/home-assistant/blob/7c155731fcdff7d8e32fd6f3fb0a7ebd45b3d7d1/homeassistant/config_entries.py#L130-L160

Reply:
It turns out there was an issue with my options update logic. https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/vizio/config_flow.py#L139-L140 sets `data` and `option` to `None`. Instead, I need to create new variables to copy into, then update those, then pass those to the `async_update_entry` call. I am submitting a separate PR to get those updated because that PR should get merged into master with #30653 whereas this can wait. Once that gets merged I will fix this PR.